relation_engine/taxa/test/config_test.py

```diff
@@ -0,0 +1,444 @@
+
+
+def test_fail_toml_error():
+    # I'm sure there's tons more possible TOML errors, but we're not here to test their code
+    # for them, are we?
+    f = BytesIO(b"input_file =  \ninput_file2 = bar")
+    _fail_config(f, ["inputfile"], False, TOMLDecodeError("Invalid value (at line 1, column 15)"))
+
+
+def test_fail_missing_input_key():
+    f = BytesIO("\n".join([
+        'input_file = "./bar.txt"',
+        'input_file2 = "./foo.txt"',
+    ]).encode("utf-8"))
+    _fail_config(f, ["inputfile"], False, ValueError("Missing input key inputfile"))
+
+
+def test_fail_missing_arango_section():
+    f = BytesIO("\n".join([
+        'input_file = "./bar.txt"',
+    ]).encode("utf-8"))
+    _fail_config(f, ["input_file"], False, ValueError("Missing section Arango"))
+
+
+def test_fail_missing_versioning_section():
+    f = BytesIO("\n".join([
+        'input_file = "./bar.txt"',
+        "[Arango]",
+        'url = "https://foo.com"',
+    ]).encode("utf-8"))
+    _fail_config(f, ["input_file"], False, ValueError("Missing section Versioning"))
+
+
+def test_fail_missing_url():
+    for url in ["", 'url = "    \t  "']:
+        f = BytesIO("\n".join([
+            'input_file = "./bar.txt"',
+            "[Arango]",
+            url,
+            'database = "mydb"',
+            "[Versioning]",
+            'load-version = "ver"',
+        ]).encode("utf-8"))
+        _fail_config(f, ["input_file"], False, ValueError(
+            "Missing value for key url in section Arango"))
+
+
+def test_fail_missing_database():
+    for db in ["", 'database = "    \t  "']:
+        f = BytesIO("\n".join([
+            'input_file = "./bar.txt"',
+            "[Arango]",
+            'url = "https://foo.com"',
+            db,
+            "[Versioning]",
+            'load-version = "ver"',
+        ]).encode("utf-8"))
+        _fail_config(f, ["input_file"], False, ValueError(
+            "Missing value for key database in section Arango"))
+
+
+def test_fail_missing_password():
+    for p in ["", 'password = "    \t  "']:
+        f = BytesIO("\n".join([
+            'input_file = "./bar.txt"',
+            "[Arango]",
+            'url = "https://foo.com"',
+            'database = "db"',
+            'user = "pwd is required if user is misssing"',
+            p,
+            "[Versioning]",
+            'load-version = "ver"',
+        ]).encode("utf-8"))
+        _fail_config(f, ["input_file"], False, ValueError(
+            "If user is present in the Arango section, password must be present"))
+
+
+def test_fail_missing_load_registry_collection():
+    for lrc in ["", 'load-registry-collection = "    \t  "']:
+        f = BytesIO("\n".join([
+            'input_file = "./bar.txt"',
+            "[Arango]",
+            'url = "https://foo.com"',
+            'database = "db"',
+            lrc,
+            "[Versioning]",
+            'load-version = "ver"',
+        ]).encode("utf-8"))
```

Comment:
I am continually pained by multi-line strings, but this seems okay given the context

Reply:
If there's a better way to do it definitely lmk